system_metrics_collector/src/topic_statistics_collector/subscriber_topic_statistics.hpp

```diff
@@ -0,0 +1,175 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#ifndef TOPIC_STATISTICS_COLLECTOR__SUBSCRIBER_TOPIC_STATISTICS_HPP_
+#define TOPIC_STATISTICS_COLLECTOR__SUBSCRIBER_TOPIC_STATISTICS_HPP_
+
+#include <chrono>
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "rclcpp_lifecycle/lifecycle_node.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+#include "received_message_age.hpp"
+#include "received_message_period.hpp"
+#include "system_metrics_collector/collector.hpp"
+#include "system_metrics_collector/metrics_message_publisher.hpp"
+
+namespace topic_statistics_collector
+{
+
+/**
+ * Class which makes periodic topic statistics measurements, using a ROS2 timer.
+ */
+template<typename T>
+class SubscriberTopicStatistics : public system_metrics_collector::MetricsMessagePublisher,
+  public rclcpp_lifecycle::LifecycleNode,
+  public system_metrics_collector::Collector
+{
+public:
```

Comment:
let's move to private what can be, and if it can't see if it could be protected? It seems a lot of methods are `public` in this class.

Reply:
Moved to private and protected wherever possible.